internal/gke/deploy.go

```diff
@@ -956,11 +954,83 @@ func createServiceAccountsIAMPolicyRole(ctx context.Context, config CloudConfig)
+		// list of regions where to rollout the app, as specified by the user.
+		return cfg.Regions[0], nil
+	}
+
+	// Check if the config cluster actually exists.
+	exists, err := hasCluster(ctx, config, region)
+	if err != nil {
+		return "", err
+	}
+	if exists {
+		return region, nil
+	}
+	// The config cluster doesn't exist; disable ingress.
+	// TODO(rgrandl): we have to force the deletion of the ingress, because some
+	// other resources that depend on the ingress might be lingering around. We
+	// should delete all these resources as well.
+	_, err = runGcloud(config,
+		fmt.Sprintf("Disable multi-cluster ingress for cluster %q in %q",
+			applicationClusterName, region), cmdOptions{},
+		"container", "fleet", "ingress", "disable", "--force")
+	if err != nil {
+		return "", err
+	}
+
+	// Set up the config cluster in the first region from the list of regions
+	// where to rollout the app, as specified by the user.
+	return cfg.Regions[0], nil
+}
+
+// getRunningConfigRegion returns the region that manages all the applications
+// within a Service Weaver project.
+//
+// Returns none if no config cluster is running.
+func getRunningConfigRegion(config CloudConfig) (string, error) {
+	// Check if there is a config cluster already running.
+	out, err := runGcloud(config, "", cmdOptions{},
+		"container", "fleet", "ingress", "describe",
+		"--format=value(spec.multiclusteringress.configMembership)")
+	if err != nil {
+		// There is no config cluster.
+		return "none", nil
+	}
+
+	// There is a config cluster entry. Extract the region where the config
+	// cluster should be running.
+	out = strings.TrimSuffix(out, "\n") // remove trailing newline
+	fName := fmt.Sprintf("projects/%s/locations/global/memberships/%s", config.Project, applicationClusterName)
+	if strings.Contains(out, fmt.Sprintf("%s-config", applicationClusterName)) {
+		// This check is for legacy deployments. We used to create a config cluster
+		// called serviceweaver-config.
+		fName = fName + "-config"
+	}
+	var region string
+	if _, err := fmt.Sscanf(out, fName+"-%s", &region); err != nil {
+		return "", fmt.Errorf("unable to parse config region name: %v", err)
+	}
+	return region, nil
+}
+
 // ensureConfigCluster sets up a Service Weaver configuration cluster, returning the
 // cluster information and the IP address of the gateway that routes ingress
 // traffic to all Service Weaver applications.
-func ensureConfigCluster(ctx context.Context, config CloudConfig, cfg *config.GKEConfig, name, region string, bindings iamBindings) (*ClusterInfo, string, error) {
-	cluster, err := ensureManagedCluster(ctx, config, cfg, name, region)
+func ensureConfigCluster(ctx context.Context, config CloudConfig, cfg *config.GKEConfig, bindings iamBindings) (*ClusterInfo, string, error) {
+	configRegion, err := getConfigRegion(ctx, config, cfg)
+	if err != nil {
+		return nil, "", err
+	}
+	cluster, err := ensureManagedCluster(ctx, config, configRegion)
```

Comment:
I'm concerned that for legacy deployments, this will create another config cluster in the application region.

Is that what we want? I'm worried that if the original config cluster was managing applications, the new cluster will start with a blank slate and things will get seriously messed up.

I think the code needs to keep the legacy config cluster and use it indefinitely, unfortunately. That's not too bad, we just need to carry the config cluster name throughout the code.